### PR TITLE
Allow creation of PlotNFTs in self-farming state

### PR DESCRIPTION
**Pull request by aqk**

### CODE REVIEW

File: chia/cmds/plotnft_funcs.py [hunk 1]
```diff
@@ -9,47 +10,68 @@
...
+
+
+async def create(args: dict, wallet_client: WalletRpcClient, fingerprint: int) -> None:
+    state = args["state"]
+
+    # Could use initial_pool_state_from_dict to simplify
+    if state == "SELF_POOLING":
+        pool_url: Optional[str] = None
+        relative_lock_height = uint32(0)
+        new_address = await wallet_client.get_next_address("1", True)
```
Comment (mariano54) [inline chia/cmds/plotnft_funcs.py:52]: Can we just not provide an address and let the wallet backend do it? That means less requests from the CLI.

File: chia/cmds/plotnft.py [hunk 2]
```diff
@@ -44,13 +44,19 @@ def get_login_link_cmd(launcher_id: str) -> None:
...
-def create_cmd(wallet_rpc_port: int, fingerprint: int, pool_url: str) -> None:
+@click.option("-u", "--pool_url", help="HTTPS host:port of the pool to join", type=str, required=False)
+@click.option("-s", "--state", help="Initial state of Plot NFT: local or pool", type=str, required=True)
+def create_cmd(wallet_rpc_port: int, fingerprint: int, pool_url: str, state: str) -> None:
     import asyncio
     from .wallet_funcs import execute_with_wallet
     from .plotnft_funcs import create
 
-    extra_params = {"pool_url": pool_url}
+    if pool_url is not None and state == "local":
```
Comment (mariano54) [inline chia/cmds/plotnft.py:54]: Case sensitive? You can use .lower()

**Review by mariano54 — APPROVED** [review]
LGTM! Minor nits. Assuming you tested both, pool and self.

File: chia/cmds/plotnft_funcs.py (hunk 1 above)
Comment (aqk) [reply chia/cmds/plotnft_funcs.py:52]: Yes, that's what I wanted to do too - the only reason I did it this way is that I was wondering about a use case like this: you want to create and control a plotnft, but you want it to pay out to a friend for some reason. (Of course, you can always redirect it later).

I admit that's a bit of a stretch, and it's easy to add later, so I think I'll change it to what you are suggesting.

File: chia/cmds/plotnft.py (hunk 2 above)
Comment (aqk) [reply chia/cmds/plotnft.py:54]: Thanks

**Pull request merged**